### ISSUE TITLE
Add option to trigger jobs (incl conversations) on de-orbiting.

### ISSUE THREAD

**Issue by petervdmeer** [endless-sky/endless-sky#6225]
### Problem Description

When I'm writing new mission-content for ES, then I sometimes have ideas for missions that should happen while the player is landing, but before the player actually has landed. Examples:
- A repulsor failure when de-orbiting for landing. (Player can choose between "Return to orbit" and "Attempt to land anyway", where the latter gives a risk of crashing/dying.
- Some ship requesting help when the player goes in for landing. (Help, our pilot passed out, can you help us land this ship?)
- The spaceport being very busy. (Player can choose to "Return to orbit" or "Try to land anyway.", the latter resulting in a fine for the player for a wrongly parked ship (outside the docks).)
- The player colliding with a satellite.
- Some sort of [https://en.wikipedia.org/wiki/D._B._Cooper](DB Cooper) jumping from the players ship.
- Storms/Fog/Winds/Planetary Weather preventing the player from landing (or making it unsafe).
- Some passenger that is very afraid during de-orbiting.
- Fire in the cargo-hold (player requesting an emergency landing)

### Related Issue Links

https://github.com/endless-sky/endless-sky/issues/5912 Trigger mission Fail at specific locations
https://github.com/endless-sky/endless-sky/issues/3297#issuecomment-347031147 Additional locations where to offer missions
https://github.com/endless-sky/endless-sky/pull/5781#issuecomment-798917554 Skydivers being too drunk or too inexperienced to jump; player can choose to let them jump, getting paid but risk of them dieing. Or choose to land with them, but also not getting paid.

### Desired Solution

Add a keyword `de-orbiting` that works similar to the `(job | landing | assisting | boarding)` keywords.
The planet-UI should show during conversations in this node, but the UI should not show the regular planet image, but an image from orbit, and most buttons (like "Spaceport", "Outfitter", etc) should be disabled.
Refuelling and repairing should also only happen after this step (for in case this step results in `leave` causing the ship not to land.

Here is how such a mission could look like:
```
mission "Repulsor failure"
	de-orbiting
	on offer
		conversation
			"You see a warning that one of the repulsors is not working on your ship."
		choice
			"(Return to orbit to fix the issue in space.)"
				leave
			"(Attempt to land anyway.)
		"You attempt to land, but your ship crashes badly."
			die
	on entry
		"You fix the repulsor by an EVA while in orbit. It should be working fine on next landing."
``

### Alternative Approaches

I also considered to have an `on de-orbit` keyword similar to the `on enter <system>` keyword and I'm not sure which option is better. And for some new missions I looked for other places where they would fit, for example just handle them on landing.
I also considered other names for the keyword;
- "atmospheric entry" (but some planets don't have atmospheres)
- "final approach" (an airport/airplanes term, might fit as well but is usually later than de-orbiting)


### Additional Context

We probably only want one on-orbit mission to be active at any time; because those on-orbit missions would typically have their conversations start in high-orbit and end near landing or with re-orbiting.

**Comment by yafflehk** [comment]
These sound fun, I don't know if the engine of the game could make the mechanical failures linked to the state of your ship. e.g. if your shields or hull are low they become more likely. 
Given that the game saves when you land on a planet, how would a de-orbit that ends in death be handled?

**Comment by petervdmeer** [comment]
Yes, we can set conditions on when a mission should be triggered and I expect that we will also use the ships hull and shields in such conditions in the future.

De-orbit happens before landing, so if you die while de-orbiting then I would expect no new savegame to be created.

**Comment by Zitchas** [comment]
So, this is basically an "on-landing" that takes place *before* the usual landing stuff takes place? (EI no refueling, repair, etc).
Sounds useful for facilitating a greater variety of mission sequences. Having a different background scene during the conversation is good, but would require a a lot more planetary scenes. Maybe we could have a dozen or so starscape pictures in the same format as the planet pictures (a few with haze) and each planet would specify one of these starscapes to be displayed during any "on de-orbit" mission triggers. Then there would be a nice view that clearly indicates one is not on-planet without needing a new picture for *every* planet. Alternatively, displaying the planet sprite set on a starscape background could probably work too.

**Comment by petervdmeer** [comment]
@Zitchas Good ideas. Your comment also gave me some more alternative ideas which I added to the description of the issue-request.

**Comment by Amazinite** [comment]
> Add a keyword `de-orbiting` that works similar to the `(job | landing | assisting | boarding)` keywords.
> I also considered to have an `on de-orbit` keyword...

Seems like both would be helpful in different scenarios. The `deorbiting` tag would simply cause the mission to offer when deorbiting, but an `on deorbit [<planet>?]` action could be useful in different scenarios.
